implementations/rust/ockam/ockam_command/src/tcp/outlet/list.rs

```diff
@@ -59,11 +61,21 @@ impl ListCommand {
 
         let (outlets, _) = try_join!(send_req, progress_output)?;
 
-        let list = opts.terminal.build_list(
-            &outlets.list,
-            &format!("Outlets on Node {}", node.node_name()),
-            &format!("No TCP Outlets found on node {}.", node.node_name()),
-        )?;
+        let list: String = {
+            let empty_message = fmt_info!(
+                "No TCP Outlets found on node {}",
+                color_primary(node.node_name())
+            );
+            match outlets.list.is_empty() {
+                true => empty_message,
+                false => opts.terminal.build_list(
+                    &outlets.list,
+                    &format!("TCP Outlets on node {}", color_primary(node.node_name())),
+                    &empty_message,
+                )?,
+            }
```

Comment:
Similarly, to avoid this match, we should change the `build_list` behavior to not print the header when the list is empty.

With these two changes to the `build_list` function, you wouldn't need to change the client code in any command and have a consistent output.

Reply:
great catch! done